Lib/subprocess.py

```diff
@@ -1307,15 +1307,15 @@ def _execute_child(self, args, executable, preexec_fn, close_fds,
                 try:
                     exception_name, hex_errno, err_msg = (
                             errpipe_data.split(b':', 2))
+                    err_msg = err_msg.decode(errors="surrogatepass")
                 except ValueError:
                     exception_name = b'SubprocessError'
                     hex_errno = b'0'
```

Comment:
Please build err_msg using errpipe_data.decode(errors="surrogatepass") to prevent b'...' from repr(bytes). Example:

err_msg = errpipe_data.decode(errors="surrogatepass")
err_msg = 'Bad exception data from child: %s' % err_msg